### ISSUE TITLE
Why wandb can only show the first step log after second step?

### ISSUE THREAD

**Issue by Yumin-Sun-00** [wandb/client#907]
### Description
I log the loss, metrics and video every epoch step.

I found that wandb will not show up the first step log immediately, it always show up after the second step. This makes the debug process longer, cause I always need to wait two epochs.

**Comment by cvphelps** [comment]
Hi Yumin, thanks for letting me know about this. How have you instrumented your code? Could you share a sample script so we can reproduce this behavior?

**Comment by Yumin-Sun-00** [comment]
> Hi Yumin, thanks for letting me know about this. How have you instrumented your code? Could you share a sample script so we can reproduce this behavior?

```
def train(self):
        """
        Full training logic
        """
        for epoch in range(self.start_epoch, self.epochs + 1):
            val_log = self._valid_epoch()
            train_log = self._train_epoch()
            result = {**train_log, **val_log}

            # Log loss and metrics to wandb
            for key, value in result.items():
                if key == 'train_metrics':
                    for m in self.metrics:
                        wandb.log({'train_' +m: result['train_metrics'][m]}, step=epoch)
                elif key == 'val_metrics':
                    for m in self.metrics:
                        wandb.log({'val_' + m: result['val_metrics'][m]}, step=epoch)
                else:
                    wandb.log({key:value}, step=epoch)

            # Log video to wandb
            with torch.no_grad():
               # some procedure to generate movie
                wandb.log({"train_movie": wandb.Video(train_movie, fps=0.5, format="gif")},step= epoch)
                wandb.log({"valid_movie": wandb.Video(valid_movie, fps=0.5, format="gif")},step= epoch)
            
            # save model
```

**Comment by vanpelt** [comment]
Hey @Yumin-Sun-00, you can add `commit=True` to your last call to `wandb.log`.  By default we only commit the data when the step changes.  By specifying commit, you're saying you're not going to add anymore keys to that step.

**Comment by ariG23498** [comment]
Hey @Yumin-Sun-00 
In the past year we've majorly reworked the CLI and UI for Weights & Biases. We're closing issues older than 6 months. Please comment to reopen.
